### PR TITLE
[RFR] React frontend bootstrap & products

### PR DESCRIPTION
**Pull request by djhi**
- [x] Updated redux and use react-router-redux
- [x] Made react-router history configurable (hashHistory in dev to avoid prbs with webpack serving apps at `http://localhost/app`, browserHistory in prod)
- [x] Bootstrapped routes pattern to keep routes with their domain
- [x] Added some twitter bootstrap structure
- [x] Added basic pages for product list and product item

product list
product item

### CODE REVIEW

File: config/development-dist.js [hunk 1]
```diff
@@ -8,6 +8,9 @@ module.exports = {
                 database: 'boilerplate',
             },
         },
+        frontend: {
+            history: 'createHashHistory',
```
Comment (alexisjanvier) [inline config/development-dist.js]: What's the difference between `createBrowserHistory` and 'createHashHistory`?

Comment (djhi) [inline config/development-dist.js]: `createHashHistory`: `http://localhost:8080/frontend#/products?_k=iqmqw2`

`createBrowserHistory`: `http://localhost:8080/frontend/products`

Comment (djhi) [inline config/development-dist.js]: In production, using `createBrowserHistory` will give us urls like:
`http://our_domain.com/products`

Comment (alexisjanvier) [inline config/development-dist.js]: Thanks :)

File: src/frontend/js/app/routes.js [hunk 2]
```diff
@@ -1,10 +1,14 @@
-import React from 'react';
-import { Route, IndexRoute } from 'react-router';
+/* globals FRONTEND_URL */
```
Comment (Kmaschta) [inline src/frontend/js/app/routes.js]: No longer necessary

File: src/frontend/js/product/ProductDetails.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import numeral from 'numeral';
+import Helmet from 'react-helmet';
+import { Link } from 'react-router';
+import ProductItem from './ProductItem';
+
+const ProductDetails = ({ product: { id, name, description, price, imageUrl }}) => (
+    <div className="row">
+        <Helmet title={`New App - ${name}`} />
```
Comment (Kmaschta) [inline src/frontend/js/product/ProductDetails.js]: Can we display the `New App` from the config ?

File: src/frontend/js/product/ProductList.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React, { PropTypes } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import Helmet from 'react-helmet';
+import ProductItem from './ProductItem';
+
+const ProductList = ({ products }) => (
+    <div className="row">
+        <Helmet title="New App - Products" />
```
Comment (Kmaschta) [inline src/frontend/js/product/ProductList.js]: Idem

**Comment by djhi** [comment]
PS: I'll review the client product model when the API will be ready

**Comment by Kmaschta** [comment]
:+1:

**Pull request merged**